luarules/gadgets/unit_minesweeper_detection.lua

```diff
@@ -5,201 +5,117 @@ function gadget:GetInfo()
-							end
-							
-						end
 
+	for sweeper, sweeperDef in pairs(minesweepers[currentTeam])  do
+		local x, y, z = Spring.GetUnitPosition(sweeper)
+		if x and z then
+			local nearUnits = Spring.GetUnitsInCylinder(x, z, minesweeperRanges[sweeperDef])
+			for _, nearUnit in ipairs(nearUnits) do
+				local nearUnitDefID = Spring.GetUnitDefID(nearUnit)
+				if currentTeam ~= Spring.GetUnitTeam(nearUnit) then
+					if mineIDs[nearUnitDefID] then
+						minesToReveal[nearUnit] = true
 					end
-				else
-					minesweepers[teamouter][miner] = nil
-					--unset , errant entry somehow
-
 				end
 			end
+		end
+	end
 
+	for mineToReveal, _ in pairs(minesToReveal) do
+		if revealedMines[currentTeam][mineToReveal] == nil then
+			revealedMines[currentTeam][mineToReveal] = true
+			--show in vision and radar, and prevent engine from immediately resseting that state
+			Spring.SetUnitLosState(mineToReveal, currentTeam, 3)
+			Spring.SetUnitLosMask(mineToReveal, currentTeam, 3)
+		end
+	end
+end
 
-		
-
-			for team in pairs(teamIDs) do
-
-				--Spring.Echo('hornet minesToReveal (team)' , team)
-				--Spring.Debug.TableEcho(minesToReveal)
-				for mineToReveal in next, (minesToReveal[team] or {}) do
-					--Spring.Echo('hornet poi2 minesToReveal loop starting, team: ' , team)
-					--Spring.Echo('mineToReveal loop,', mineToReveal)
-					if revealedMines[team][mineToReveal] == nil then
-						--show 
-						--Spring.Echo('revealedMines[team][mineToReveal] = mineToReveal', team, mineToReveal)
-						revealedMines[team][mineToReveal] = mineToReveal
-						Spring.SetUnitLosState(mineToReveal, team, 3) --show in vision and radar
-						Spring.SetUnitLosMask(mineToReveal, team, 3) -- prevent engine from immediately resetting that state
-
-					end
-				end
-					
-
-				--Spring.Echo('hornet revealedMines (team)' , team)
-				--Spring.Debug.TableEcho(revealedMines)
-				if next(revealedMines[team]) ~= nil then
-
-					--Spring.Echo('hornet minesToReveal (team)' , team)
-					--Spring.Debug.TableEcho(minesToReveal)
-
-					--if any mines -are- uncloaked that no longer should be, recloak
-					--if table.count(minesToReveal[team]) > 0 then
-					if next(minesToReveal[team]) ~= nil then
-
-						for revealedMine in pairs(revealedMines[team]) do
-							if minesToReveal[team][revealedMine] == nil then
-								--reset this team mask
-								Spring.SetUnitLosMask(revealedMine, team, 0)
-								revealedMines[team][revealedMine] = nil
-							end
-						end
-					else
-						--if empty and previously decloaked not empty, reset all mines
-						for revealedMine in pairs(revealedMines[team]) do
-							Spring.SetUnitLosMask(revealedMine, team, 0)
-							revealedMines[team][revealedMine] = nil
-						end
-					end
-
-				end
+local function processRevealedMines(teamID)
+	for revealedMine, _ in pairs(revealedMines[teamID]) do
+		Spring.SetUnitLosMask(revealedMine, teamID, 0)
+		revealedMines[teamID][revealedMine] = nil
+	end
+end
 
+if #teamIDs + 1 >= 30 then
```

Comment:
Game.gameSpeed

Reply:
why?

Reply:
because that is the self-documenting version of 30

Reply:
do i go `local GAMESPEED = Game.gameSpeed` and them use that or do i just go `Game.gameSpeed` several times?